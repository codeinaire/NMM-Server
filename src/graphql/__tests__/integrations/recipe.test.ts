import {createTestClient} from 'apollo-server-testing';
import gql from 'graphql-tag';

import { constructTestServer } from '../__utils';
import { mockRecipe } from '../resolvers/recipe.query.test';

// the mocked SQL DataSource store
const mockStore = {
  recipes: {
    findAllRecipes: jest.fn(),
  }
};

const GET_RECIPES = gql`
  query GET_RECIPES {
    recipes {
      id
      title
      ingredients
      method
      hashtags
      attribution {
        name
        url
      }
      standardResolution {
        width
        height
        url
      }
    }
  }
`

describe('Queries', () => {
  // TODO - findAllRecipes isn't mocking, fix later
  it('finds all recipes', async () => {
    const { server, recipeAPI } = constructTestServer({
      context: () => ({
        event: {
          headers: 'test headers'
        }
      })
    });

    console.log('SERVER', server);



    recipeAPI.store = mockStore;
    recipeAPI.store.recipes.findAllRecipes.mockReturnValueOnce({get: () => ({mockRecipe})});

    console.log('recipeAPI.store', recipeAPI);
    const { query } = createTestClient(server);
    const res = await query({
      query: GET_RECIPES,
    });
    console.log('RESPONSE', res);

    expect(res).toMatchSnapshot();
  });
});

describe('Mutations', () => {
});
