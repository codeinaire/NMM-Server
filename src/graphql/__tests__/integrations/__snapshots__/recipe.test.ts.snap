// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries finds all recipes 1`] = `
Object {
  "data": Object {
    "recipes": null,
  },
  "errors": Array [
    [GraphQLError: this.store.recipes.findAll is not a function],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
